tests/test_resampler.py

```diff
@@ -0,0 +1,46 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Tests to verify that the resampler can be used successfully from Python."""
+
+import datetime as dt
+
+from frequenz_resampling import Resampler, ResamplingFunction
+
+
+def test_resampler() -> None:
+    """Test the resampler."""
+    start = dt.datetime(1970, 1, 1, tzinfo=dt.timezone.utc)
+    step = dt.timedelta(seconds=1)
+    resampler = Resampler(dt.timedelta(seconds=5), ResamplingFunction.Average, 1, start)
+
+    for i in range(10):
+        resampler.push(start + i * step, i + 1)
+
+    expected = [
+        (start + 5 * step, 3.0),
+        (start + 10 * step, 8.0),
+    ]
+
+    resampled = resampler.resample(start + 10 * step)
+
+    assert resampled == expected
+
+
+def test_resampling_none() -> None:
```

Comment:
Probably add one test per resampling function to make sure the enum values are correct too? :laughing: